tools/harvest_source_import/remote_ckan/lib.py

```diff
@@ -101,11 +107,45 @@ def get_full_harvest_source(self, hs):
             logger.error(error)
             self.errors.append(error)
             # yield incomplete version
+            self.save_temp_json('harvest-source', hs['name'], hs)
             return hs
         else:
             full_hs = response.json()
             self.harvest_sources[hs['name']] = full_hs['result']
+            self.save_temp_json('harvest-source', full_hs['result']['name'], full_hs['result'])
             return full_hs['result']
+    
+    def get_full_organization(self, org):
+        """ get full info (including job status) for a Harvest Source """
+        organization_show_url = f'{self.url}/api/3/action/organization_show'
+        org_id = org.get('name', org.get('id', None))
+        if org_id is None:
+            return None
+        
+        params = {'id': org_id}
+        headers = self.get_request_headers(include_api_key=False)
+        
+        logger.info(f'Get organization data {organization_show_url} {params}')
+        
+        # not sure why require this
+        headers['Content-Type'] = 'application/x-www-form-urlencoded'
+        params = json.dumps(params)
+        response = requests.post(organization_show_url, data=params, headers=headers)
+        
+        if response.status_code >= 400:
+            error = f'Error [{response.status_code}] getting organization info:\n {params} \n{response.headers}\n {response.content}'
+            logger.error(error)
+            self.errors.append(error)
+            # yield incomplete version
+            self.save_temp_json('organization', org.get('name', org.get('id', 'unnamed')), org)
+            self.organizations[org['name']] = org
+            return org
+        else:
+            response = response.json()
+            org = response['result']
+            self.save_temp_json('organization', org.get('name', org.get('id', 'unnamed')), org)
```

Comment:
Line 146-148 are dups of 140-142 ... this could be refactored slightly to avoid the duplication.

Reply:
Yes, you are right. Fixed [here](https://github.com/GSA/catalog.data.gov/pull/78/commits/42ef6d80f0604512ae273a1419e32c03b8742d66)